modules/stroeerCoreBidAdapter.js

```diff
@@ -0,0 +1,249 @@
+
+    return win.document.getElementById(elId);
+  };
+
+  const visibleInWindow = (el, win) => {
+    const rect = el.getBoundingClientRect();
+    const inView = (rect.top + rect.height >= 0) && (rect.top <= win.innerHeight);
+
+    if (win !== win.parent) {
+      return inView && visibleInWindow(win.frameElement, win.parent);
+    }
+
+    return inView;
+  };
+
+  try {
+    return visibleInWindow(resolveElement(elementId), utils.getWindowSelf());
+  } catch (e) {
+    // old browser, element not found, cross-origin etc.
+  }
+  return undefined;
+}
+
+function buildUrl({host: hostname = DEFAULT_HOST, port = DEFAULT_PORT, securePort, path: pathname = DEFAULT_PATH}) {
+  if (securePort) {
+    port = securePort;
+  }
+
+  return utils.buildUrl({protocol: 'https', hostname, port, pathname});
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  isBidRequestValid: (function () {
+    const validators = [];
+
+    const createValidator = (checkFn, errorMsgFn) => {
+      return (bidRequest) => {
+        if (checkFn(bidRequest)) {
+          return true;
+        } else {
+          utils.logError(`invalid bid: ${errorMsgFn(bidRequest)}`, 'ERROR');
+          return false;
+        }
+      }
+    };
+
+    function isBanner(bidReq) {
+      return (!bidReq.mediaTypes && !bidReq.mediaType) ||
+        (bidReq.mediaTypes && bidReq.mediaTypes.banner) ||
+        bidReq.mediaType === BANNER;
+    }
+
+    validators.push(createValidator((bidReq) => isBanner(bidReq),
+      bidReq => `bid request ${bidReq.bidId} is not a banner`));
+    validators.push(createValidator((bidReq) => typeof bidReq.params === 'object',
+      bidReq => `bid request ${bidReq.bidId} does not have custom params`));
+    validators.push(createValidator((bidReq) => utils.isStr(bidReq.params.sid),
+      bidReq => `bid request ${bidReq.bidId} does not have a sid string field`));
+    validators.push(createValidator((bidReq) => bidReq.params.ssat === undefined || [1, 2].indexOf(bidReq.params.ssat) > -1,
+      bidReq => `bid request ${bidReq.bidId} does not have a valid ssat value (must be 1 or 2)`));
+
+    return function (bidRequest) {
+      return validators.every(f => f(bidRequest));
+    }
+  }()),
+
+  buildRequests: function (validBidRequests = [], bidderRequest) {
+    const anyBid = bidderRequest.bids[0];
+
+    const bidRequestWithSsat = find(validBidRequests, bidRequest => bidRequest.params.ssat);
+    const bidRequestWithYl2 = find(validBidRequests, bidRequest => bidRequest.params.yl2);
+
+    const payload = {
+      id: bidderRequest.auctionId,
+      bids: [],
+      ref: getTopWindowReferrer(),
+      ssl: isSecureWindow(),
+      mpa: isMainPageAccessible(),
+      timeout: bidderRequest.timeout - (Date.now() - bidderRequest.auctionStart),
+      ssat: bidRequestWithSsat ? bidRequestWithSsat.params.ssat : 2, // TODO
+      yl2: bidRequestWithYl2 ? bidRequestWithYl2.params.yl2 : (localStorage.sdgYieldtest === '1'), // TODO
```

Comment:
What about these parameters? (`yl2`, `ssat`, `ab`) Are they required in the official/public Stroeer adapter? Do we need to send them? Do we need to compute the values dynamically, or can we set it to e. g. `2` all the times?

Reply:
Good spotting. Yes, `yl2`, `ssat` and `ab` can be removed. The backend will use correct defaults.

Also the code related to `bidResponse.bidPriceOptimisation` can be removed.